diff --git a/packer_templates/pkr-builder.pkr.hcl b/packer_templates/pkr-builder.pkr.hcl
index 5f3b027..06c514a 100644
--- a/packer_templates/pkr-builder.pkr.hcl
+++ b/packer_templates/pkr-builder.pkr.hcl
@@ -50,6 +50,7 @@ locals {
                 "${path.root}/scripts/${var.os_name}/sudoers_${var.os_name}.sh",
                 "${path.root}/scripts/${var.os_name}/systemd_${var.os_name}.sh",
                 "${path.root}/scripts/${var.os_name}/hyperv_${var.os_name}.sh",
+                "${path.root}/scripts/hardening.sh",
                 "${path.root}/scripts/${var.os_name}/cleanup_${var.os_name}.sh",
                 "${path.root}/scripts/_common/parallels_post_cleanup_debian_ubuntu.sh"
                 ] : (
@@ -144,6 +145,12 @@ build {
     pause_after = "10s"
     except      = var.is_windows ? local.source_names : null
   }
+
+  provisioner "file" {
+    sources     = ["${path.root}/config/ansible.cfg", "${path.root}/config/local.yml"]
+    destination = "/tmp/"
+  }
+
   # Run OS specific scripts
   provisioner "shell" {
     environment_vars  = local.nix_environment_vars
