diff --git a/packer_templates/pkr-builder.pkr.hcl b/packer_templates/pkr-builder.pkr.hcl
index 780ffe3..38e544e 100644
--- a/packer_templates/pkr-builder.pkr.hcl
+++ b/packer_templates/pkr-builder.pkr.hcl
@@ -50,6 +50,7 @@ locals {
                 "${path.root}/scripts/${var.os_name}/networking_${var.os_name}.sh",
                 "${path.root}/scripts/${var.os_name}/sudoers_${var.os_name}.sh",
                 "${path.root}/scripts/${var.os_name}/systemd_${var.os_name}.sh",
+                "${path.root}/scripts/hardening.sh",
                 "${path.root}/scripts/${var.os_name}/hyperv_${var.os_name}.sh",
                 "${path.root}/scripts/${var.os_name}/cleanup_${var.os_name}.sh",
                 "${path.root}/scripts/_common/parallels_post_cleanup_debian_ubuntu.sh"
@@ -127,6 +128,13 @@ build {
     scripts           = local.common_scripts
     except            = var.is_windows ? local.source_names : null
   }
+
+  # Run Ansible playbook
+  provisioner "file" {
+    sources     = ["${path.root}/config/ansible.cfg", "${path.root}/config/local.yml"]
+    destination = "/tmp/"
+  }
+
   # Run OS specific scripts
   provisioner "shell" {
     environment_vars  = local.nix_environment_vars
diff --git a/packer_templates/pkr-sources.pkr.hcl b/packer_templates/pkr-sources.pkr.hcl
index b318d93..815b75d 100644
--- a/packer_templates/pkr-sources.pkr.hcl
+++ b/packer_templates/pkr-sources.pkr.hcl
@@ -187,7 +187,6 @@ locals {
         ["modifyvm", "{{.Name}}", "--usb-xhci", "on"],
         ["modifyvm", "{{.Name}}", "--mouse", "usb"],
         ["modifyvm", "{{.Name}}", "--keyboard", "usb"],
-        ["storagectl", "{{.Name}}", "--name", "IDE Controller", "--remove"],
       ]
     )
   ) : var.vboxmanage
